orderable/templates/admin/edit_inline/orderable_tabular.html

```diff
@@ -19,12 +19,14 @@
 $(document).ready(function() {
 
     var order_array = [];
+    var min_value = 0;
     $("#{{ inline_admin_formset.formset.prefix }}-group .tabular.inline-related tbody tr.has_original td.field-sort_order :input").each(function() {
-            var id = $(this).val();
-            order_array.push(id);
-            $(this).parent().append("<span id='{{ inline_admin_formset.formset.prefix }}-neworder-" + id + "' class='sorthandle'>Order</span>");;
-            $(this).hide();
-        });
+        var id = Math.max(min_value, $(this).val());  // Ensure that the values increase.
+        min_value = id + 1;
```

Comment:
Am I right in saying that this will change the value of every element the second time items are re-ordered?

Reply:
Sort of ... if two items have the same value then, yes reordering will result in them being ordered ins such a way that no values are equal, so any items that need to be incremented will be incremented.

The old/original version simply moves the `sort_order` values about. The problem with this approach is that `sort_order` values that are equal do not change.

Reply:
Given items a, b, c, with sort order 3, 5, 5:

| Item | Sort Order | 
|---|---|
| a | 3 | 
| b | 5 |
| c | 5 |

With the old version reordering b and c has no effect as it swapps `5` for `5`. WIth the new version `c` gets incremented to `6` and then swapped with `5`

| Item | Sort Order | 
|---|---|
| a | 3 | 
| c | 5 |
| b | 6 |

Reply:
> subsequent times simply swap the values (as before).

That's what I was concerned about. Sounds like you're on top of it.